File: library/src/main/java/org/incendo/interfaces/transform/AppliedTransform.java
package org.incendo.interfaces.transform;

import org.incendo.interfaces.pane.Pane;
import org.incendo.interfaces.properties.Trigger;
import org.incendo.interfaces.view.InterfaceView;

import java.util.Set;

public record AppliedTransform<P extends Pane>(
    int priority,
    Set<Trigger> triggers,
    Transform<P> transform,
    boolean async
) implements Transform<P> {

    @Override
    public void apply(final P pane, final InterfaceView view) {
        this.transform.apply(pane, view);
    }

}

File: library/src/main/java/org/incendo/interfaces/transform/AppliedTransform.kt
package org.incendo.interfaces.transform

import org.incendo.interfaces.next.pane.Pane
import org.incendo.interfaces.next.properties.Trigger

public class AppliedTransform<P : Pane>(
    internal val priority: Int,
    internal val triggers: Set<Trigger>,
    transform: Transform<P>
) : Transform<P> by transform
